core/tee/tee_rpmb_fs.c

```diff
@@ -2111,25 +2065,15 @@ static int rpmb_fs_write(TEE_Result *errno, int fd, const void *buf,
 	uint32_t start_addr;
 
 	if (!size)
-		return 0;
+		return TEE_SUCCESS;
 
 	mutex_lock(&rpmb_mutex);
 
-	if (!buf) {
```

Comment:
No, the caller should be able to avoid calling with pointers as doing that is usually quite useless, calling with a NULL pointer is usually due to some fault in the code.
If there's inconsistencies it's probably because I've missed removing it.